stringify.go

```diff
@@ -0,0 +1,220 @@
+package gcfg
+
+import (
+	"encoding"
+	"fmt"
+	"math/big"
+	"reflect"
+	"strconv"
+	"strings"
+	"unicode"
+	"unicode/utf8"
+)
+
+// Stringify returns the ini format representation of `config`.
+func Stringify(config interface{}) (string, error) {
+	configPtr := reflect.ValueOf(config)
+	if configPtr.Kind() != reflect.Ptr || configPtr.Elem().Kind() != reflect.Struct {
+		return "", fmt.Errorf("Config must be a pointer to a struct")
+	}
+	configValue := configPtr.Elem()
+
+	var s string
+	for i := 0; i < configValue.NumField(); i++ {
+		fieldValue := configValue.Field(i)
+		fieldStruct := configValue.Type().Field(i)
+		if !fieldStruct.IsExported() {
+			continue
+		}
+
+		iniFieldName := iniKey(fieldStruct)
+
+		if fieldValue.Kind() == reflect.Struct {
+			res, err := stringifyStructFields(fieldValue)
+			if err != nil {
+				return "", err
+			}
+
+			s += fmt.Sprintf("[%s]\n%s\n", iniFieldName, res)
+		} else if fieldValue.Kind() == reflect.Map {
+			if fieldStruct.Type.Key().Kind() != reflect.String {
+				return "", fmt.Errorf("The map keys must to be of string type, instead they are of %s type", fieldStruct.Type.Key().Kind())
+			}
+
+			if fieldStruct.Type.Elem().Kind() == reflect.String {
```

Comment:
We just checked this above so we can remove this if statement

Reply:
This above was checking for the key, this one is checking for the value.